library/choose_disk.py

```diff
@@ -0,0 +1,455 @@
+#!/usr/bin/python
+
+from ansible.module_utils.basic import AnsibleModule
+import json
+import logging
+import os
+import re
+import subprocess
+
+DOCUMENTATION = '''
+---
+module: choose_disk
+author: Erwan Velu <erwan@redhat.com>
+short_description: Choose disks based on their features
+description:
+    To be completed
+'''
+
+logger = logging.getLogger('choose_disk')
+module = None
+
+
+def _equal(left, right):
+    return left == right
+
+
+def _gt(left, right):
+    return float(left) > float(right)
+
+
+def _lt(left, right):
+    return float(left) < float(right)
+
+
+_REGEXP = re.compile(r'^([^(]+)'          # function name
+                     r'\(\s*([^,]+)'      # first argument
+                     r'(?:\s*,\s*(.+))?'  # remaining optional arguments
+                     r'\)$')              # last parenthesis
+
+
+def convert_units(value):
+    ''' Convert units to ease comparaison '''
+    value = str(value).lower().strip()
+    storage_units = {
+            'kb': 1024,
+            'kib': 1000,
+            'mb': 1024*1024,
+            'mib': 1000*1000,
+            'gb': 1024*1024*1024,
+            'gib': 1000*1000*1000,
+            'tb': 1024*1024*1024*1024,
+            'tib': 1000*1000*1000*1000,
+            'pb': 1024*1024*1024*1024*1024,
+            'pib': 1000*1000*1000*1000*1000
+    }
+
+    # Units are storage units
+    for size in storage_units.keys():
+        if value.endswith(size):
+            real_value, unit = value.split(" ")
+            return str(float(real_value) * storage_units[unit])
+
+    return value
+
+
+def get_keys_by_ceph_order(physical_disks):
+    '''
+    Return a list of keys where ceph disks are reported first
+    while keeping the list sorted
+    '''
+    ceph_disks = []
+    non_ceph_disks = []
+    for physical_disk in sorted(physical_disks):
+        if "ceph_prepared" in physical_disks[physical_disk]:
+            ceph_disks.append(physical_disk)
+        else:
+            non_ceph_disks.append(physical_disk)
+
+    return ceph_disks + non_ceph_disks
+
+
+def find_match(physical_disks, lookup_disks):
+    ''' Find a set of matching devices in physical_disks
+    '''
+    matched_devices = {}
+    ignored_devices = []
+
+    logger.info("Looking for matches")
+    # Inspecting every disk we search for
+    for disk in sorted(lookup_disks):
+
+        current_lookup = dict(lookup_disks[disk])
+        infinite = False
+        current_type = ""
+        if "infinite" in current_lookup.keys():
```

Comment:
there is no need to use `keys()` here since `item in dict` will look at a dict's keys